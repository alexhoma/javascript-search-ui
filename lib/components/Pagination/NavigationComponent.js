"use strict";
exports.__esModule = true;
var preact_1 = require("preact");
var Template_1 = require("../Template");
/**
 * Arrow navigation component
 */
function NavigationComponent(_a) {
    var isVisible = _a.isVisible, classNames = _a.classNames, template = _a.template, handleClick = _a.handleClick;
    return (isVisible)
        ? (preact_1.h("li", { className: classNames, onClick: handleClick },
            preact_1.h(Template_1["default"], { template: template, dictionary: this.props.dictionary })))
        : null;
}
exports["default"] = NavigationComponent;
