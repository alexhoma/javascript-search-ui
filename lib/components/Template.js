"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
exports.__esModule = true;
var preact_1 = require("preact");
var Hogan = require("hogan.js");
/**
 * Template
 */
var Template = /** @class */ (function (_super) {
    __extends(Template, _super);
    function Template() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        /**
         * Render template
         *
         * @param template
         * @param result
         *
         * @return {any}
         */
        _this.renderTemplate = function (template, result) {
            /**
             * Compile template using hogan.js
             */
            var compiledTemplate = Hogan.compile(template);
            var output = compiledTemplate.render(result);
            return {
                __html: output
            };
        };
        return _this;
    }
    /**
     * Render
     *
     * @return {any}
     */
    Template.prototype.render = function () {
        var props = this.props;
        var template = props.template;
        var data = props.data;
        var className = props.className;
        return (template)
            ? preact_1.h("div", { className: className, dangerouslySetInnerHTML: this.renderTemplate(template, data) })
            : null;
    };
    return Template;
}(preact_1.Component));
exports["default"] = Template;
