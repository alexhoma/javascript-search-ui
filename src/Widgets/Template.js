/** @jsx h */
import { h, Component } from 'preact';
import Hogan from 'hogan.js';

class Template extends Component {
    renderTemplate = (template, data) => {
        let compiledTemplate = Hogan.compile(template);
        let output = compiledTemplate.render(data);


        return {
            __html: output
        };
    };

    render() {
        const {
            template,
            data,
            className
        } = this.props;

        return (template)
            ? <div
                className={className}
                dangerouslySetInnerHTML={this.renderTemplate(template, data)}
            />
            : null
        ;
    }
}

export default Template;