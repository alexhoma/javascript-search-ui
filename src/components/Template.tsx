import { h, Component } from 'preact';
import {TemplateProps} from "./TemplateProps";
import * as Mustache from 'mustache';

/**
 * Template
 */
class Template extends Component<TemplateProps> {

    /**
     * Render template
     *
     * @param template
     * @param result
     * @param dictionary
     *
     * @return {any}
     */
    renderTemplate = (template, result, dictionary) => {

        let trans = function() {
            return function(text, render) {
                return dictionary[text] ?? text;
            }
        }

        let output = Mustache.render(template, {...result, ...{
            'trans': trans
        }});

        return {
            __html: output
        };
    };

    /**
     * Render
     *
     * @return {any}
     */
    render() {

        const props = this.props;
        const template = props.template;
        const data = props.data;
        const className = props.className;
        const dictionary = props.dictionary ?? {};

        return (template)
            ? <div
                className={className}
                dangerouslySetInnerHTML={this.renderTemplate(template, data, dictionary)}
            />
            : null
        ;
    }
}

export default Template;
