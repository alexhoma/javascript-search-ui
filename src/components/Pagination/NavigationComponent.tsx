import {h} from 'preact';
import Template from "../Template";

/**
 * Arrow navigation component
 */
function NavigationComponent({
    isVisible,
    classNames,
    template,
    handleClick
}) {
    return (isVisible)
        ? (
            <li
                className={classNames}
                onClick={handleClick}
            >
                <Template
                    template={template}
                    dictionary={this.props.dictionary}
                />
            </li>
        )
        : null
    ;
}

export default NavigationComponent;
