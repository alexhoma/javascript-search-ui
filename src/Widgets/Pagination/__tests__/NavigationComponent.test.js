/**
 * @jsx h
 */
import {h} from 'preact';
import {deep} from 'preact-render-spy';
import NavigationComponent from './../NavigationComponent';
import Template from '../../Template';

describe('<NavigationComponent />',() => {
    const defaultProps = {
        classNames:`navigation-component-class`,
        template: `(((Template)))`,
    };

    it('should render if isVisible is set to true', () => {
        let tree = getNavigationComponent({
            ...defaultProps,
            isVisible: true
        });

        expect(tree).toMatchSnapshot();
    });
    it('should NOT render if isVisible is set to false', () => {
        let tree = getNavigationComponent({
            ...defaultProps,
            isVisible: false
        });

        expect(tree).toMatchSnapshot();
    });
    it('should handleClick()', () => {
        let mock = jest.fn();
        let tree = getNavigationComponent({
            ...defaultProps,
            isVisible: true,
            handleClick: mock
        });

        tree.find('.navigation-component-class').at(0).simulate('click');

        expect(mock).toBeCalled();
    });


    const getNavigationComponent = ({
        isVisible,
        classNames,
        template,
        handleClick
    }) => {
        return deep(
            <NavigationComponent
                isVisible={isVisible}
                classNames={classNames}
                template={template}
                handleClick={handleClick}
            />
        );
    }
});