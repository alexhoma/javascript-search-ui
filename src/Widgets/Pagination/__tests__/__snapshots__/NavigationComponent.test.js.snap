// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationComponent /> should NOT render if isVisible is set to false 1`] = `
preact-render-spy (1 nodes)
-------
{null}

`;

exports[`<NavigationComponent /> should render if isVisible is set to true 1`] = `
preact-render-spy (1 nodes)
-------
<li class="navigation-component-class">
  <div>(((Template)))</div>
</li>

`;
